libs/data/bbhnet/data/utils.py

```diff
@@ -66,14 +69,31 @@ def sample_kernels(
     # always make sure that the center of x's
     # 1st axis is in the kernel that we sample
     # if we're doing >1D sampling
-    min_sample_start = max(
-        x.shape[-1] // 2 + 1 - trigger_distance_size - (size / 2), 0
-    )
-    max_sample_start = min(
-        x.shape[-1] // 2 - 1 - (size / 2) + trigger_distance_size,
-        x.shape[-1] - size,
-    )
 
+    # TODO: not sure if theres a logical way to
+    # combine this if else statement.
+    # Was spending too much time thinking about it
+
+    # allows for kernels that don't contain trigger
+    if trigger_distance_size > 0:
+        min_sample_start = max(
+            x.shape[-1] // 2 + 1 - trigger_distance_size - size, 0
+        )
+        max_sample_start = min(
+            x.shape[-1] // 2 - 1,
+            x.shape[-1] - size,
+        )
+
+    # trigger must be trigger_distance_size away from
+    # either kernel edge
+    elif trigger_distance_size <= 0:
+        min_sample_start = max(
+            x.shape[-1] // 2 + 1 - trigger_distance_size - size, 0

```

Comment:
This should be `+ trigger_distance_size` right? Because that way the leftmost edge will always be  `< size` away from `t0`

Reply:
So If i remember correctly its a `+` because trigger_distance_size is negative in this instance. Does that make sense to you? If not I'll re think this.